### PR TITLE
expose argv[0] as `$env.PROCESS_PATH`

### LINKED ISSUE #11059: Expose `argv[0]`
### Related problem

I would like `argv[0]` ie the name the script was invoked with, exposed in scripts

### Describe the solution you'd like

`argv[0]` is available as the `path` argument here:

https://github.com/nushell/nushell/blob/77a1c3c7b2f3a110d48bcb792968e6b0d85d4bb7/crates/nu-cli/src/eval_file.rs#L19

The information is lost here:

https://github.com/nushell/nushell/blob/77a1c3c7b2f3a110d48bcb792968e6b0d85d4bb7/crates/nu-cli/src/eval_file.rs#L34-L44

https://github.com/nushell/nushell/blob/77a1c3c7b2f3a110d48bcb792968e6b0d85d4bb7/crates/nu-cli/src/eval_file.rs#L97-L100

### Describe alternatives you've considered

`$"./($env.CURRENT_FILE | path relative-to (pwd))"`, but this is not necessarily the same thing as `argv[0]` (an obvious exception is windows)

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by p00f**
closes #11059 

# Description
I'm not sure what the consensus was after discussing this in discord, so I'm creating a PR as suggested

# User-Facing Changes

TBD
# Tests + Formatting
TBD

# After Submitting
TBD

### CODE REVIEW

**Comment by p00f** [comment]
@fdncred @sophiajt

**Comment by p00f** [comment]
```
/tmp/workspaces/0PMKZ: ^cat foo
#!/home/chinmay/stuff/forks/nushell/target/debug/nu

def main [] {
    printf $"I was called as ($env.ORIGINAL_CMDLINE)"
}
/tmp/workspaces/0PMKZ: ./foo
I was called as ./foo
```

**Comment by p00f** [comment]
even if we don't reach a consensus on this, i'd like to at least fix this: https://github.com/nushell/nushell/issues/11059#issuecomment-1811866262, so pointers would be appreciated

**Comment by fdncred** [comment]
ORIGINAL_CMDLINE infers the command and all the parameters, is that how it works?

**Comment by p00f** [comment]
No, it's only the name the script was invoked with (like `argv[0]` in C)

This is why ORIGINAL_CMDLINE is a bad name

**Comment by fdncred** [comment]
ok, I don't really like ORIGINAL_CMDLINE so I'm throwing out some others.

- PROGRAM_NAME
- EXE_NAME
- EXECUTABLE
- PROGRAM_FILENAME
- LAUNCH_CMD
- PROCESS_PATH
- ARGV0

I'm kind of leaning towards PROCESS_PATH, any takers?

**Comment by p00f** [comment]
Same

**Comment by sholderbach** [comment]
When the name bikeshedding is done let's update the title to make the commit better discoverable.

File: crates/nu-cli/src/eval_file.rs [hunk 1]
```diff
@@ -98,6 +98,10 @@ pub fn evaluate_file(
         "CURRENT_FILE".to_string(),
         Value::string(file_path.to_string_lossy(), Span::unknown()),
     );
+    stack.add_env_var(
+        "ORIGINAL_CMDLINE".to_string(),
```
Comment (fdncred) [inline crates/nu-cli/src/eval_file.rs:102]: Let's try process path if that's ok. We can always change it later if we don't like that either. I'll land it if you accept the changes.
```suggestion
        "PROCESS_PATH".to_string(),
```

**Comment by p00f** [comment]
@sholderbach does the pr title look good?

**Comment by sholderbach** [comment]
Thanks! 👍

**Comment by p00f** [comment]
also, when a script has a `def main`,
```
> ./tools/build.nu
Error: nu::parser::missing_positional

  × Missing required positional argument.
   ╭─[<commandline>:1:1]
 1 │ main
   ·     ▲
   ·     ╰── missing src
   ╰────
  help: Usage: main <src> <dest> . Use `--help` for more information.
```
the error message (last line) should also use this process path instead of `main`.

Where is the code for this?

**Comment by fdncred** [comment]
Let's handle the "main" part in another PR. My suggestion would be to search for "main" in the code and MissingPositional to see where they're used.

Seems like this is what's being called. You may be able to track it down from here.
https://github.com/nushell/nushell/blob/5d283755e3cfdf8c040776fe598f253d1a120157/crates/nu-protocol/src/parse_error.rs#L341-L347

**Comment by amtoine** [comment]
@p00f 
i've updated the PR description (a bit too late), can you confirm it's correct?

**Comment by p00f** [comment]
@amtoine fixed it

**Pull request merged**